### DISCUSSION TITLE
Support for EntraID Groups for RBAC control

### DISCUSSION THREAD

**Discussion by rickschwabe** [Azure/ipam#291] [Ideas]
Hello everyone,
it would be good if access for the admins could also be controlled centrally via an Entra ID group instead of individual users.
Are there already plans to support the use of groups?

**Comment by DCMattyG** [comment]
Hi @rickschwabe, I can definitely add this to the roadmap. Shouldn't be too difficult to implement, but will likely require some changes to the App Registration to send Group information in the token.

**Reply by rickschwabe** [reply]
Hi @DCMattyG, is there already an update for this?

**Comment by gummigroda** [comment]
Hey @DCMattyG, will this be implemented in  v3.7.0 ? 🙏

**Reply by DCMattyG** [reply]
Hi @gummigroda and @rickschwabe, the short answer is "no", this will not be in v3.7.0. Why? It has to do with how can the Azure IPAM application understand group membership to begin with. Your Entra ID token, which Azure IPAM uses for authentication, does not by default have Group Claims included. That means that unless the App Registrations are updated and Group Claims are added, Azure IPAM won't be able to determine what groups users are a member of to them allow/disallow access.

So, there are a few steps that need to be taken. First, I would need to add the ability to list Entra ID Groups to assign access (in addition to Users & Principals today). Second, we would need a way to walk customers through adding the Group Claims to the Azure IPAM tokens for processing.

Since I was focused on some pretty major code changes in v3.7.0 (swapping out the Inovua React Data Grid since it's now a defunct/abandoned project) I wanted to make sure that was all handled first, then I can explore some backlogged feature requests for v3.8.0. I'll make sure to include this on my todo list there.

Apologies it's taking so long to get that feature in there, but hopefully we'll have it in there for you soon!

**Reply by gummigroda** [reply]
I understand, and thanks for all the efforts so far.
